client/components/header-cake/back.jsx

```diff
@@ -21,52 +22,60 @@ const HIDE_BACK_CRITERIA = {
 	characterLength: 8
 };
 
-export default React.createClass( {
-	displayName: 'HeaderCakeBack',
-
-	propTypes: {
+class HeaderCakeBack extends Component {
+	static propTypes = {
 		onClick: PropTypes.func,
 		href: PropTypes.string,
 		text: PropTypes.string,
-		spacer: PropTypes.bool
-	},
+		spacer: PropTypes.bool,
+	};
+
+	static defaultProps = {
+		spacer: false,
+		disabled: false,
+	};
 
-	getDefaultProps() {
-		return {
-			spacer: false,
-			disabled: false
-		};
-	},
+	state = {
+		windowWidth: viewport.getWindowInnerWidth(),
+	};
 
 	componentDidMount() {
 		this.resizeThrottled = throttle( this.handleWindowResize, 100 );
 		window.addEventListener( 'resize', this.resizeThrottled );
-	},
+	}
 
 	componentWillUnmount() {
 		window.removeEventListener( 'resize', this.resizeThrottled );
-	},
+	}
 
-	handleWindowResize() {
-		this.forceUpdate();
-	},
+	handleWindowResize = () => {
+		this.setState( {
+			windowWidth: viewport.getWindowInnerWidth(),
+		} );
+	}
 
 	hideText( text ) {
-		const windowWidth = viewport.getWindowInnerWidth();
-
 		if (
-			windowWidth <= HIDE_BACK_CRITERIA.windowWidth &&
+			this.state.windowWidth <= HIDE_BACK_CRITERIA.windowWidth &&
 			text.length >= HIDE_BACK_CRITERIA.characterLength ||
-			windowWidth <= 300
+			this.state.windowWidth <= 300
 		) {
 			return true;
 		}
 
 		return false;
-	},
+	}
 
 	render() {
-		const { text = i18n.translate( 'Back' ), href, onClick, spacer, icon } = this.props;
+		const {
+			href,
+			icon,
+			onClick,
+			spacer,
+			text,
+			translate,
+		} = this.props;
+		const backText = text || translate( 'Back' );
```

Comment:
This might be a small issue. Is it possible to achieve no text like this? A literal space would probably work, but ideally we should be able to say `text={ '' }` or some other falsey things.

An default assignment value might be best.

Reply:
I've updated it to handle empty string titles, see 832bc80bc3

Reply:
👍  nice. 🚢